In [7]:
# Load important module

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [8]:
#Loading Datasets

reviews_file = r"C:\Users\BENSON KAMAU\OneDrive\Dokumente\Projo\New folder\British Airline Reviews Analysis\ba_reviews.csv"

countries_file = r"C:\Users\BENSON KAMAU\OneDrive\Dokumente\Projo\New folder\British Airline Reviews Analysis\Countries.csv"

ba_reviews = pd.read_csv(reviews_file)

countries = pd.read_csv(countries_file)







In [17]:
#Overview of Data

print("BA Reviews Datasets:")
print(ba_reviews.info())
print("\nHead of Dataset:")
print(ba_reviews.head())



BA Reviews Datasets:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1324 entries, 0 to 1323
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   header               1324 non-null   object
 1   author               1324 non-null   object
 2   date                 1324 non-null   object
 3   place                1324 non-null   object
 4   content              1324 non-null   object
 5   aircraft             1324 non-null   object
 6   traveller_type       1323 non-null   object
 7   seat_type            1324 non-null   object
 8   route                1324 non-null   object
 9   date_flown           1324 non-null   object
 10  recommended          1324 non-null   object
 11  trip_verified        1324 non-null   object
 12  rating               1324 non-null   int64 
 13  seat_comfort         1324 non-null   int64 
 14  cabin_staff_service  1324 non-null   int64 
 15  food_beverages       1324 non-null

The dataset contains 1,324 reviews with the following key columns:

- **header, content**: The title and detailed content of the review.
- **author, place, date**: Reviewer’s details and when the review was posted.
- **aircraft, traveller_type, seat_type, route, date_flown**: Flight details including aircraft type, type of traveler, class of seat, route, and date flown.
- **recommended, trip_verified**: Whether the reviewer recommended the airline and if the trip is verified.
- **rating**: Overall rating of the experience.
- **seat_comfort, cabin_staff_service, food_beverages, ground_service, value_for_money, entertainment**: Sub-ratings for specific aspects of the experience.

In [16]:
print("Countries Dataset")
print(countries.info())
print("\nHead of Dataset:")
print(countries.head())


Countries Dataset
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 251 entries, 0 to 250
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Country    251 non-null    object
 1   Code       250 non-null    object
 2   Continent  251 non-null    object
 3   Region     251 non-null    object
dtypes: object(4)
memory usage: 8.0+ KB
None

Head of Dataset:
          Country Code Continent           Region
0     Afghanistan  AFG      Asia    Southern Asia
1   Åland Islands  ALA    Europe  Northern Europe
2         Albania  ALB    Europe  Southern Europe
3         Algeria  DZA    Africa  Northern Africa
4  American Samoa  ASM   Oceania        Polynesia


The dataset contains 1,324 reviews with the following key columns:

- **header, content**: The title and detailed content of the review.
- **author, place, date**: Reviewer’s details and when the review was posted.
- **aircraft, traveller_type, seat_type, route, date_flown**: Flight details including aircraft type, type of traveler, class of seat, route, and date flown.
- **recommended, trip_verified**: Whether the reviewer recommended the airline and if the trip is verified.
- **rating**: Overall rating of the experience.
- **seat_comfort, cabin_staff_service, food_beverages, ground_service, value_for_money, entertainment**: Sub-ratings for specific aspects of the experience.
